### PR TITLE
SA1125 tests and bug fixes

### LINKED ISSUE #300: Diagnostic unit tests: UseShorthandForNullableTypes
Implement unit tests for #58.

### LINKED ISSUE #386: SA1125 UseShorthandForNullableTypes incorrectly reported in typeof()
SA1125 is currently being reported for the following statement.

```csharp
Type nullableType = typeof(Nullable<>);
```

### LINKED ISSUE #636: SA1125 UseShorthandForNullableTypes incorrectly reported for static access through Nullable<int>
The following expression incorrectly results in SA1125:

```csharp
bool equal = Nullable<int>.Equals(null, null);
```

### LINKED ISSUE #637: SA1125 UseShorthandForNullableTypes incorrectly reported in nameof expression
The following expression incorrectly results in SA1125:

```csharp
string name = nameof(Nullable<int>);
```

### LINKED ISSUE #638: SA1125 UseShorthandForNullableTypes incorrectly reported for member in XML comment
The following expression incorrectly results in SA1125:

```csharp
/// <seealso cref="Nullable{T}.Value"/>
```

### PR DESCRIPTION
**Pull request by sharwell**
* Implement tests for SA1125 (UseShorthandForNullableTypes) and enable it by default
* Fixes #300
* Fixes #386
* Fixes #636
* Fixes #637
* Fixes #638

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1125UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        /// <summary>
+        /// This is a regression test for DotNetAnalyzers/StyleCopAnalyzers#385.
+        /// <see href="https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/385">SA1125
+        /// UseShorthandForNullableTypes incorrectly reported in XML comment</see>
+        /// </summary>
+        [Theory]
+        [InlineData("Nullable{T}", "Nullable{T}")]
+        [InlineData("System.Nullable{T}", "System.Nullable{T}")]
+        [InlineData("global::System.Nullable{T}", "global::System.Nullable{T}")]
+        public async Task TestSeeAlsoNullable(string longForm, string shortForm)
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1125UnitTests.cs]: Is there a reason why you included shortForm here? It isnt used in the test (there are a few occurences of this in this file)

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1125UnitTests.cs]: I wasn't sure if I was going to rewrite them using `[MemberData]`. It turned out there wasn't much duplication, so I can go ahead and simplify these.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1125UnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+        {{
+            var nullableValue = default({0});
+        }}
+    }}
+}}
+";
+            string testCode = string.Format(template, longForm);
+            string fixedCode = string.Format(template, shortForm);
+
+            if (testCode != fixedCode)
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1125UnitTests.cs]: This is always true.

**Comment by pdelvo** [comment]
Otherwise this change looks good :+1:

**Pull request merged**